aws/resource_aws_network_interface_test.go

```diff
@@ -77,10 +78,11 @@ func TestAccAWSENI_basic(t *testing.T) {
 		CheckDestroy:  testAccCheckAWSENIDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccAWSENIConfig,
+				Config: testAccAWSENIConfig(),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckAWSENIExists("aws_network_interface.bar", &conf),
 					testAccCheckAWSENIAttributes(&conf),
+					testAccCheckAWSENIAvailabilityZone("data.aws_availability_zones.available", "names.0", &conf),
```

Comment:
Since we do `ImportStateVerify` testing with this resource, we can skip the custom `TestCheckFunc` here (sigh, some of this old resource testing before the days of the standard SDK functions!) and instead just check against the state value via:

```go
resource.TestCheckResourceAttrPair("aws_network_interface.bar", "availability_zone", "data.aws_availability_zones.available", "names.0"),
```

Reply:
The `network_interface` resource doesn't have an `availability_zone` attribute so using `resource.TestCheckResourceAttrPair()` doesn't seem to work.

Reply:
Ha fancy that. 😅  We could always add one (since the information is readily available), but certainly outside the scope of this PR. Ignore this!